### DISCUSSION TITLE
STEP 200

### DISCUSSION THREAD

**Discussion by 7sorkr** [ponoor/step-series-support#23] [Q&A] [answered]
I'm going to connect the STEP200.
Is there anything I should refer to other than the documents on the homepage?
Or is there any change?

Thank you in advance.

**Comment by kanta** [comment, marked as answer]
So far there is no information other than [the doc page](https://ponoor.com/en/docs/step100-200/).
Just in case: The table of contents is at the bottom of that page.

If you have any clarification or missing information, please let me know. This is a good chance for me to update the documents.

**Reply by 7sorkr** [reply]
I have read several documents on the website.

Some things are overseas
I'll try it when the supplies arrive next week.
I'm a newbie, so if there's any blockage, I'll ask.

Q) Does it work only with Arduino Zero?

thank you

**Reply by kanta** [reply]
Yes, as I wrote [here](https://ponoor.com/en/docs/step100-200/get-the-materials/), possible options are;
- Arduino Zero.
- Arduino M0
- [Adafruit METRO M0 Express](https://www.adafruit.com/product/3505)

Technically Arduino UNO can work with X-Nucleo series and Ethernet shield but STEP200 sketch is too big for Arduino UNO's memory size.

**Reply by 7sorkr** [reply]
Thanks for the link and reply.

<Preparation>
- Arduino zero
- X-NUCLEO-IHM02A1
- Ethernet Shield

No problem..?

**Reply by kanta** [reply]
Yes, you can start to work with those 3 boards!

**Reply by 7sorkr** [reply]
**1-1. I'm inquiring about changing the dIp switch in STEP200. Do I just need to change the numbers in line 288 and 324 in the image?**

![DIP_Switch](https://user-images.githubusercontent.com/32606306/185199797-cbb53b68-58f2-4060-bb39-6bb731824ae3.PNG)

**Reply by 7sorkr** [reply]
**1-2. How do I configure the pins after adding 3 lines to boardsDef.h to make it a DIP switch in STEP200?**
```
    #define HAVE_DIP_SW
    #define DIP_SW_DIGITS   4
    const uint8_t dipSwPin[DIP_SW_DIGITS] = { A1, A3, A4, A5 };
```

DIP 1 ON = A1
DIP 1 OFF = GND
DIP 2 ON = A3
DIP 2 OFF = GND
DIP 3 ON = A4
DIP 3 OFF = GND
DIP 4 ON = A5
DIP 4 OFF = GND

!! Should I connect the resistance? 
Should I connect before going to A1? 
(How many ohms should I connect each? 10kohm?!?)

**Reply by 7sorkr** [reply]
**2. Is the connection method of two HOME sensors correct?**

STEP 200, **HOME Sensor A** (SW=J2(1pin))
Analog 5v pin  ---5v-----------> + (671a) 
J2 (1Pin SW)------Sensor------> out (671a)
J2 (2Pin GND)----GND--------> - (671a)

STEP 200, **HOME Sensor B** (SW=J2(5pin))
Analog 5v pin  ---5v-----------> + (671a)
J2 (5Pin SW)------Sensor------> out (671a)
J2 (6Pin GND)----GND--------> - (671a)

**Reply by 7sorkr** [reply]
**3. Is the connection method of the two LIMIT sensors correct?**

STEP200, **LIMIT Sensor A** (SW=Digital 6pin)
Analog 5v pin  ---5v-----------> + (671a)
Digital 6Pin -----Sensor------> out (671a)
J2 (4Pin GND)----GND--------> - (671a)

STEP200, **LIMIT Sensor B** (SW=Digital 7pin)
Analog 5v pin  ---5v-----------> + (671a)
Digital 7Pin -----Sensor------> out (671a)
J8 (4Pin GND)----GND--------> - (671a)

**Reply by 7sorkr** [reply]
**4. How do I connect two HOME sensors and two LIMIT sensors in the STEP200?**

**Reply by 7sorkr** [reply]
**5. L6470 SCK Movement Inquiry (SB34-->SB12)**
Do I just have to make SB12 short circuit after SB34 breakage or resistance removal as shown in the document?

**Reply by 7sorkr** [reply]
kantaさん、本当にすごいです。

And these days, I've been rack-mounting STEP400
I'll upload it when it's done

**Reply by kanta** [reply]
> **1-1. I'm inquiring about changing the dIp switch in STEP200. Do I just need to change the numbers in line 288 and 324 in the image?**
> 
> ![DIP_Switch](https://user-images.githubusercontent.com/32606306/185199797-cbb53b68-58f2-4060-bb39-6bb731824ae3.PNG)

Line 288 is part of the STEP 100 definition. You only need to change line 324.

**Reply by kanta** [reply]
> **1-2. How do I configure the pins after adding 3 lines to boardsDef.h to make it a DIP switch in STEP200?**
> 
> ```
>     #define HAVE_DIP_SW
>     #define DIP_SW_DIGITS   4
>     const uint8_t dipSwPin[DIP_SW_DIGITS] = { A1, A3, A4, A5 };
> ```
> 
> DIP 1 ON = A1 DIP 1 OFF = GND DIP 2 ON = A3 DIP 2 OFF = GND DIP 3 ON = A4 DIP 3 OFF = GND DIP 4 ON = A5 DIP 4 OFF = GND
> 
> !! Should I connect the resistance? Should I connect before going to A1? (How many ohms should I connect each? 10kohm?!?)

You don't need to connect resistors, since those pins will be pulled up in the MCU.
You can configure like this;

| state | pin |  |
| --- | --- | --- |
| ON | close | connect to the GND |
| OFF | open |  no connection, just leave it. |

**Reply by kanta** [reply]
> **2. Is the connection method of two HOME sensors correct?**
> 
> STEP 200, **HOME Sensor A** (SW=J2(1pin)) Analog 5v pin ---5v-----------> + (671a) J2 (1Pin SW)------Sensor------> out (671a) J2 (2Pin GND)----GND--------> - (671a)
> 
> STEP 200, **HOME Sensor B** (SW=J2(5pin)) Analog 5v pin ---5v-----------> + (671a) J2 (5Pin SW)------Sensor------> out (671a) J2 (6Pin GND)----GND--------> - (671a)

You are right! Only one 5V pin is available on Arduino boards so you need to split into two lines.

**Reply by kanta** [reply]
> **4. How do I connect two HOME sensors and two LIMIT sensors in the STEP200?**

You can add these lines to the board definition, as you did for the DIP switch. Pin 5 and 6 are just example, you can change any unused pins.

```c++
#define HAVE_LIMIT_GPIO
const uint8_t limitSwPin[2] = { 5, 6 };
```

**Reply by kanta** [reply]
> **5. L6470 SCK Movement Inquiry (SB34-->SB12)** Do I just have to make SB12 short circuit after SB34 breakage or resistance removal as shown in the document?

It depends on how you did "SB34 breakage". Just make sure the electrically disconnected for SB34 pad.

**Reply by kanta** [reply]
> kantaさん、本当にすごいです。
> 
> And these days, I've been rack-mounting STEP400 I'll upload it when it's done

ありがとうございます！
The rack-mount STEP400 sounds serious. Looking forward to see the results, thank you!

**Comment by 7sorkr** [comment]
STEP200  

OK - Firmware 1.1.2 (SAMD 1.8.11)
Fail - Firmware 1.2.0 (SAMD 1.8.11 & 1.8.13)
OK - Firmware 1.2.1 (SAMD 1.8.11)

![1ACD0961-2DDC-48A2-8DDD-55AFB0EC5425](https://user-images.githubusercontent.com/32606306/185699819-9454a6fb-ad6d-4b44-a0fe-c4f2a61967bf.jpeg)

**Reply by 7sorkr** [reply]
Thanks to kanta, we succeeded.

https://youtube.com/shorts/2eQnDDI1I0o?feature=share

**Reply by kanta** [reply]
Congrats,  glad to know you made it!!
